src/classes/UTIL_MasterSchedulableHelper.cls

```diff
@@ -185,4 +185,23 @@ public without sharing class UTIL_MasterSchedulableHelper {
             ERR_Handler.processError(ex, ERR_Handler_API.Context.STTG);
         }
     }
+
+    /*******************************************************************************************************
+    * @description Checks for presently running jobs of a particular method
+    * @param methodName The name of the method to check for running jobs on
+    * @return Boolean
+    */
+    public static Boolean hasRunningJob(String className) {
+        className = (className.contains('__') ? className.split('__')[1] : className);
+        Integer jobCount = [SELECT COUNT()
+            FROM AsyncApexJob
+            WHERE ApexClass.Name = :className
+            AND Status NOT IN ('Queued','Aborted','Completed','Failed')
```

Comment:
I thought about this more last night. I like this query, however there is one potential problem I see. If for some reason, a User clicks the button to manually recalculate rollups on a single Account, Contact, GAU or Recurring Donation the related batch jobs will run. And, if that button is clicked at around the same time the scheduled jobs are running, it'll block the scheduled jobs. It's a slim use case, but I don't think it's totally impossible. It could also happen in an integration if an Opp is inserted on an Account that has more than 250 attached Opps (because the Skew Mode job will be kicked off).

I think the fix here is to add a simple condition to this query for `CreatedDate < :tenMinutesAgo` with a simple calc for `DateTime tenMinutesAgo = DateTime.Now().addMinutes(-10);`. The "10" minutes part is arbitrary and I'm open to a different time frame if you think more or less time is warranted.